torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -1641,6 +1493,8 @@ std::vector<Val*> Index::getGlobalProducerStridedIndices(
       } else {
         strided_inds[i] = strided_ind;
       }
+      strided_inds[i] = GpuLower::current()->commonIndexMap().hoistIndex(
+          strided_inds[i], loops);
```

Comment:
I think we should take the sum of `strided_inds` first, and hoist the resulting val. Done in https://github.com/csarofeen/pytorch/pull/2234.